File: android/core/data/src/main/java/com/app2/core/data/remote/dto/OrderDTOs.kt
package com.app2.core.data.remote.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class OrderDTO(
    val id: Int,
    @SerialName("order_number") val orderNumber: String? = null,
    @SerialName("supplier_id") val supplierId: Int? = null,
    @SerialName("supplier_name") val supplierName: String? = null,
    @SerialName("warehouse_id") val warehouseId: Int? = null,
    val status: String? = "brouillon",
    val total: Double? = null,
    val notes: String? = null,
    @SerialName("sent_at") val sentAt: String? = null,
    @SerialName("received_at") val receivedAt: String? = null,
    @SerialName("created_at") val createdAt: String? = null
)

@Serializable
data class OrderItemDTO(
    val id: Int,
    @SerialName("order_id") val orderId: Int,
    @SerialName("product_id") val productId: Int? = null,
    @SerialName("product_name") val productName: String? = null,
    val quantity: Int? = null,
    @SerialName("unit_price") val unitPrice: Double? = null,
    @SerialName("received_qty") val receivedQty: Int? = null
)

@Serializable
data class OrderCreateRequest(
    @SerialName("supplier_id") val supplierId: Int,
    @SerialName("warehouse_id") val warehouseId: Int? = null,
    val notes: String? = null
)

@Serializable
data class OrderUpdateRequest(
    val status: String? = null,
    val notes: String? = null
)

File: ios/StockPro/Data/DTOs/OrderDTOs.swift
import Foundation

struct OrderDTO: Decodable, Identifiable {
    let id: Int
    let order_number: String?
    let supplier_id: Int?
    let supplier_name: String?
    let warehouse_id: Int?
    let warehouse_name: String?
    let status: String
    let total: Double?
    let notes: String?
    let created_at: String
    let received_at: String?

    var statusLabel: String {
        switch status {
        case "brouillon": "Brouillon"
        case "recue": "Reçue"
        case "payee": "Payée"
        case "annulee": "Annulée"
        default: status
        }
    }
}

struct OrderItemDTO: Decodable, Identifiable {
    let id: Int
    let product_id: Int
    let product_name: String?
    let quantity: Int
    let price: Double
    let total: Double?
}

struct OrderCreateRequest: Encodable {
    let supplier_id: Int
    let warehouse_id: Int
    let notes: String?
    let items: [OrderItemRequest]
}

struct OrderItemRequest: Encodable {
    let product_id: Int
    let quantity: Int
    let price: Double
}

struct OrderUpdateRequest: Encodable {
    let status: String
    let notes: String?
}
